service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureVmStep.java

```diff
@@ -177,9 +177,22 @@ public StepResult undoStep(FlightContext context) throws InterruptedException {
     ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
 
     try {
+      VirtualMachine resolvedVm =
+              computeManager
+                      .virtualMachines()
+                      .getByResourceGroup(
+                              azureCloudContext.getAzureResourceGroupId(), resource.getVmName());
+
       computeManager
```

Comment:
Is this delete idempotent? We don't want any exceptions coming from undo if it's already deleted.